src/pcf857x.ts

```diff
@@ -230,9 +239,14 @@ export abstract class PCF857x<PinNumber extends PCF8574.PinNumber | PCF8575.PinN
       // decrease the use count of the GPIO and unexport it if not used anymore
       this._gpio['pcf857xUseCount']--;
       if (this._gpio['pcf857xUseCount'] === 0) {
+        // error-proof in case somehow a negative value for gpioPin was somehow allowed during enableInterrupt() call
+        if (this._gpioPin >= 0) {
+          // nullify the registered gpio from our allInstancesUsedGpios array as count is 0 and it is being unexported
+          PCF857x._allInstancesUsedGpios[this._gpioPin] = null;
+        }
         this._gpio.unexport();
       }
-
+      this._gpioPin = -1;
```

Comment:
```ts
this._gpioPin = null;
```

Reply:
Agreed.

Reply:
As we've already decorated the gpio object with a use count, I'm wondering does it make sense to add a decorated 'pcf857xGpioPin' field to the gpio object or maintain the gpioPin on each individual pcf857x instance?

I will update the PR with the original way and if you want, we could move it onto the gpio object.